### PR TITLE
[bcl] Optimize Process.GetCurrentProcess ().MainModule.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

**Comment by DavidKarlas** [comment]
@monojenkins build pkg

File: mcs/class/System/System.Diagnostics/Process.cs [hunk 1]
```diff
@@ -98,7 +100,14 @@ private struct ProcInfo
...
 		public ProcessModule MainModule {
 			get {
-				return(this.Modules[0]);
+				/* Optimize Process.GetCurrentProcess ().MainModule */
+				if (processId == NativeMethods.GetCurrentProcessId ()) {
+					if (current_main_module == null)
+						current_main_module = this.Modules [0];
+					return current_main_module;
+				} else {
+					return this.Modules [0];
```
Comment (akoeplinger) [inline mcs/class/System/System.Diagnostics/Process.cs]: why aren't we caching `this.Modules [0]` in all cases? is there a way it could change?

Comment (vargaz) [reply mcs/class/System/System.Diagnostics/Process.cs]: Because we are caching it in a static variable, so we can only store 1 value.

Comment (akoeplinger) [reply mcs/class/System/System.Diagnostics/Process.cs]: ahh, I see 😄

**Comment by DavidKarlas** [comment]
FYI: This speeded up opening simple(just create from template) solution with 1 project in MonoDevelop from ~8.8sec to ~6.2sec.

**Pull request merged**